assets/js/googlesitekit/modules/datastore/modules.js

```diff
@@ -236,47 +221,19 @@ export const baseControls = {
 	[ REFETCH_AUTHENTICATION ]: createRegistryControl( ( { dispatch } ) => () => {
 		return dispatch( CORE_USER ).fetchGetAuthentication();
 	} ),
-	[ WAIT_FOR_MODULES ]: createRegistryControl( ( registry ) => ( { payload: {} } ) => {
-		// Select first to ensure resolution is always triggered.
-		const { getModules, hasFinishedResolution } = registry.select( STORE_NAME );
-		getModules();
-		const modulesAreLoaded = () => hasFinishedResolution( 'getModules', [] );
-		if ( modulesAreLoaded() ) {
-			return;
-		}
-		return new Promise( ( resolve ) => {
-			const unsubscribe = registry.subscribe( () => {
-				if ( modulesAreLoaded() ) {
-					unsubscribe();
-					resolve();
-				}
-			} );
-		} );
-	} ),
 };
 
 const baseReducer = ( state, { type, payload } ) => {
 	switch ( type ) {
 		case REGISTER_MODULE: {
 			const { slug, settings } = payload;
-			const { modules: existingModules } = state;
-			const defaults = {
-				description: null,
-				icon: null,
-				order: 10,
-				homepage: null,
-				internal: false,
-				active: false,
-				connected: false,
-				name: slug,
-			};
+
 			return {
 				...state,
-				modules: {
-					...existingModules,
+				clientDefinitions: {
+					...state.clientDefinitions,
 					[ slug ]: {
-						...defaults,
-						...( existingModules?.[ slug ] || {} ),
+						...( state.clientDefinitions[ slug ] || {} ),
```

Comment:
Instead of this, now that we have this cleaner approach (where client and server are separated), let's also restrict module registration to be unique per slug. Similar to `registerWidget`, the reducer should trigger a console warning and not actually register the module if it is already registered on the client.